src/Microsoft.Extensions.SecretManager.Tools/Internal/MsBuildProjectReader.cs

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.Extensions.ProjectModel.Internal;
+using GracefulException = Microsoft.Extensions.SecretManager.Tools.Internal.GracefulException;
+
+namespace Microsoft.Extensions.SecretManager.Tools.Internal
+{
+    internal class MsBuildProjectReader
+    {
+        public string ReadUserSecretId(IFileInfo fileInfo)
```

Comment:
Do you need an `IFileInfo` or could you just use `Stream` instead (and maybe keep this overload too)? It makes testing easier